### ISSUE TITLE
Events are not working

### ISSUE THREAD

**Issue by juanpenarandaosf** [Polymer/lit-element#743]
### Description
The custom events are not being received by the function set in the attribute. As seen in the following screenshot, the event is said to be emitted but it never executes the given function, which would log 'event received'. I also tested setting `on-my-event`, but it did not work either.

![image](https://user-images.githubusercontent.com/48287642/61571186-5ff0bd00-aa57-11e9-8a5c-31bbc134dd38.png)

I am following the instructions in here https://lit-element.polymer-project.org/guide/events#custom-events

#### Live Demo
https://stackblitz.com/edit/fcrpih?file=index.html

#### Steps to Reproduce
1. Create an element
1. Dispatch an event from the element, like `my-event`
1. In the HTML side, define `onMyEvent` attribute for the tag and set a function
1. Event is not reaching the defined `onMyEvent` attribute

#### Expected Results
The event calls the function defined in the `onMyEvent` attribute

#### Actual Results
The function defined in the `onMyEvent` attribute is never called

### Browsers Affected

- [x] Chrome
- [ ] Firefox
- [ ] Edge
- [ ] Safari 11
- [ ] Safari 10
- [ ] IE 11

### Versions
- lit-element: v2.2.0
- webcomponents: v2.2.10

**Comment by juanpenarandaosf** [comment]
Closing issue, as it works with `addEventListener`, in the following way

```html
<script>
  const el = document.getElementById('my-greeting');
  el.addEventListener('my-event', (e) => {
    console.log('received', e.detail);
  });
</script>
```

![image](https://user-images.githubusercontent.com/48287642/61641747-4c1aa600-ac65-11e9-8d6c-a20959e97a2e.png)
_Event being received_

**Comment by justinfagnani** [comment]
@juanpenarandaosf the event syntax uses a `@` prefix, not an `on` prefix:

```ts
html`
  <simple-greeting @my-event=${(e) => console.log(e)}></simple-greeting>
`
```

**Comment by dwojtyca** [comment]
Can we re-open this issue? According to the docs, (https://lit-element.polymer-project.org/guide/events) lit-element based events should be handled with `onMyEvent="(e) => { console.log(e) }"` but it doesn't work for me also. 

@justinfagnani syntax that you write about is to use inside lit-element based component, but inside regular HTML `on` prefix should be used and all dashes should be changed to camel case.

**Comment by juanpenarandaosf** [comment]
> @juanpenarandaosf the event syntax uses a `@` prefix, not an `on` prefix:
> 
> ```ts
> html`
>   <simple-greeting @my-event=${(e) => console.log(e)}></simple-greeting>
> `
> ```

Thank you @justinfagnani, but I understand that `@` sintax is used from inside lit-element and in this case I am trying to access it from outside with plain HTML and JS

**Comment by juanpenarandaosf** [comment]
> Can we re-open this issue? According to the docs, (https://lit-element.polymer-project.org/guide/events) lit-element based events should be handled with `onMyEvent="(e) => { console.log(e) }"` but it doesn't work for me also.
> 
> @justinfagnani syntax that you write about is to use inside lit-element based component, but inside regular HTML `on` prefix should be used and all dashes should be changed to camel case.

Yes, I believe we can reopen this issue if you have also reproduced it. Please, keep in mind, though, that you can use `addEventListener` to overcome the issue, so you don't get stuck if you are working in something

**Comment by LarsDenBakker** [comment]
You will need to open an issue to the HTML spec then. If you're working in plain HTML, you're not using `lit-html`'s templating system. the `@` syntax is just sugar for `addEventListener`.

If you want this to work:
```
<div onMyEvent="(e) => console.log(e)>
```

You will need to implement an attribute which uses `eval` to convert the string to a function since you can only sets strings in html. Native implementations of event handlers (`onclick`, `onload` etc.) are basically the same, and considered a bad practice nowadays since they need to be defined for each separate event.

**Comment by justinfagnani** [comment]
I filed an issue to remove the section in docs that mentions `onMyEvent=`. That's not something we've ever supported. I think it's trying to show how you might use a LitElement in _another_ system, like Angular or Vue, but it doesn't exactly match their syntaxes either.
